### PR TITLE
Add support for aws_ec2_vpc_net role in Linchpin

### PR DESCRIPTION
**Pull request by samvarankashyap**
$subject
example:
```yaml
aws-ec2-vpc-net:
  topology:
    topology_name: ec2vpc
    resource_groups:
      - resource_group_name: "awsvpc"
        resource_group_type: "aws"
        resource_definitions:
          - name: demodayvpc
            role: aws_ec2_vpc_net
            region: us-east-1
            cidr_block: 10.10.0.0/16
            tags:
              module: aws_ec2_vpc_net
              this: works
            tenancy: dedicated
        credentials:
          filename: aws.yaml
          profile: default
```
fixes: #827 
should be merged after #894

### CODE REVIEW

File: docs/source/aws.rst [hunk 1]
```diff
@@ -96,6 +96,16 @@ AWS Security Groups can be provisioned using this resource.
    action. This is because other resources may depend on the now existing
    resource.
 
+aws_ec2_vpc_net
+---------------
+
+AWS VPC networks can be provisioned using this resource.
+
+* :docs1.5:`Topology Example <workspace/topologies/aws-ec2-vpc-net.yml>`
```
Comment (14rcole) [inline docs/source/aws.rst]: should this be `workspaces/aws/topologies/aws-ec2-vpc-net.yml`?

**Comment by samvarankashyap** [comment]
[test]

File: docs/source/aws.rst (hunk 1 above)
Comment (samvarankashyap) [reply docs/source/aws.rst]: aws folder doesnot have a topologies folder however, the above file is referenced as example . For tests workspaces/aws/Pinfile is being used

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
rebased to fix conflicts with develop branch

**Comment by samvarankashyap** [comment]
@14rcole @p3ck : Needs approval

File: docs/source/aws.rst (hunk 1 above)
Comment (p3ck) [reply docs/source/aws.rst]: @samvarankashyap I don't understand your reply to @14rcole 

Where in this commit is there a file workspace/topologies/aws-ec2-vpc-net.yml?

Comment (samvarankashyap) [reply docs/source/aws.rst]: @p3ck Its in docs/source/examples/workspaces/aws/topologies/aws-ec2-vpc-net.yml 
my bad @14rcole was correct . I need to modify the path in docs.

**Comment by p3ck** [comment]
the unit tests are failing because we don't have gcc. :-(

We will need to add gcc to the container

**Comment by samvarankashyap** [comment]
@p3ck How ome it worked before then ??

**Comment by samvarankashyap** [comment]
@p3ck : we do have gcc in all the dockerfiles
https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/config/Dockerfiles/centos7/Dockerfile

**Comment by p3ck** [comment]
The unit tests are executed via HDSL in the ansible-executor.  We may have to actually provision a resource to run it.

I don't know why it worked before.  Did you add/update any Requires in the latest patch?

**Comment by samvarankashyap** [comment]
@p3ck : I added boto3 in requirements.txt. However , That's in previous commits which worked fine for unit tests.

**Comment by samvarankashyap** [comment]
[test]

**Pull request merged**